src/controller-v5.sol

```diff
@@ -0,0 +1,375 @@
+
+    uint256 public split = 500;
+    uint256 public constant max = 10000;
+
+    function initialize(
+        address _governance,
+        address _strategist,
+        address _timelock,
+        address _devfund,
+        address _treasury
+    ) public initializer {
+        governance = _governance;
+        strategist = _strategist;
+        timelock = _timelock;
+        devfund = _devfund;
+        treasury = _treasury;
+    }
+
+    function setDevFund(address _devfund) public {
+        require(msg.sender == governance, "!governance");
+        devfund = _devfund;
+    }
+
+    function setTreasury(address _treasury) public {
+        require(msg.sender == governance, "!governance");
+        treasury = _treasury;
+    }
+
+    function setStrategist(address _strategist) public {
+        require(msg.sender == governance, "!governance");
+        strategist = _strategist;
+    }
+
+    function setSplit(uint256 _split) public {
+        require(msg.sender == governance, "!governance");
+        require(_split <= max, "numerator cannot be greater than denominator");
+        split = _split;
+    }
+
+    function setOneSplit(address _onesplit) public {
+        require(msg.sender == governance, "!governance");
+        onesplit = _onesplit;
+    }
+
+    function setGovernance(address _governance) public {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function setTimelock(address _timelock) public {
+        require(msg.sender == timelock, "!timelock");
+        timelock = _timelock;
+    }
+
+    function setJar(address _token, address _jar) public {
+        require(
+            msg.sender == strategist || msg.sender == governance,
+            "!strategist"
+        );
+        require(jars[_token] == address(0), "jar");
+        jars[_token] = _jar;
+    }
+
+    function approveJarConverter(address _converter) public {
+        require(msg.sender == governance, "!governance");
+        approvedJarConverters[_converter] = true;
+    }
+
+    function revokeJarConverter(address _converter) public {
+        require(msg.sender == governance, "!governance");
+        approvedJarConverters[_converter] = false;
+    }
+
+    function approveStrategy(address _token, address _strategy) public {
+        require(msg.sender == timelock, "!timelock");
+        approvedStrategies[_token][_strategy] = true;
+    }
+
+    function revokeStrategy(address _token, address _strategy) public {
+        require(msg.sender == governance, "!governance");
+        require(strategies[_token] != _strategy, "cannot revoke active strategy");
+        approvedStrategies[_token][_strategy] = false;
+    }
+
+    function setConvenienceFee(uint256 _convenienceFee) external {
+        require(msg.sender == timelock, "!timelock");
+        convenienceFee = _convenienceFee;
+    }
+
+    function setStrategy(address _token, address _strategy) public {
```

Comment:
if strategist is an EOA, what happens in the case his keys are compormised? ist there time to react from the pickle team in case compromised strategist calls this with a full rug strategy?

Our approach is to limit new code deployed and attached to governance only

Reply:
strategist is a multisig, not EOA. So I think it's not the case.